src/core/vector/qgsvectorlayer.cpp

```diff
@@ -1402,7 +1402,7 @@ QgsGeometry::OperationResult QgsVectorLayer::splitFeatures( const QgsCurve *curv
 int QgsVectorLayer::addTopologicalPoints( const QgsGeometry &geom )
 {
   if ( !isValid() || !mEditBuffer || !mDataProvider )
-    return -1;
+    return 1;
```

Comment:
Indeed, this can be considered as an error. However, there is a slighty difference between the error considered in `QgsVectorLayerEditUtils::addtopologicalPoints` and this one.

QgsVectorLayerEditUtils returns 1 if the geometry is not spatial or null. This class will not perform the addTopologicalPoints if the layer is invalid or not editable, so it's a different error for me.

For this method, one could have theses returns:

- -1 in case of **layer** error (invalid or non editable)
- 0 in case of success
- 1 in case of **geometry** error (non spatial or null geometry)
- 2 in case no vertices needed to be added

Reply:
That makes sense. Adjusted in https://github.com/qgis/QGIS/pull/43854/commits/92231d74c861f6d40997efbe1e3a4a6de8f849fa